File: nysol_python/mcmd/methods/convTex/db/smp/mmvstats.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mmvstats'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,value
1,5
2,1
3,3
4,4
5,4
6,6
7,1
8,4
9,7
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
移動窓の合計を計算する。
最初の行は期数に満たないため出力されない。
'''
script['sh_code']='''
mmvstats s=id f=value t=2 c=sum i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mmvstats(s="id", f="value", t="2", c="sum", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mmvstats.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,value
1,5
2,1
3,3
4,4
5,4
6,6
7,1
8,4
9,7
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Calculate sum of sliding window.
The first row is not printed as there is less than the required nubmer of intervals for computation.
EOF
scp=<<'EOF'
more dat1.csv
mmvstats s=id f=value t=2 c=sum i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)
